### PR TITLE
input.conf: rotate with r and t, move subs with ctrl+up and down

### PR DESCRIPTION
**Pull request by guidocella**
Rotating videos is a commonly needed feature and the command do it is not obvious, so bind it by default. r and t are chosen because:

- r is the first letter of rotate
- t is the third and fifth letter of rotate
- They are next to each other, so it is intuitive that the left key rotates left and the right key rotates right

Move add sub-pos to ctrl+up and down, because it was hard to remember which of r and t moves up and which moves down, while with arrow bindings it is obvious, and because ctrl{,+shift}+{left,right} already have sub bindings.

### CODE REVIEW

File: etc/input.conf [hunk 1]
```diff
@@ -119,9 +121,8 @@
 #Alt+1 set current-window-scale 1.0     # reset the window size
 #Alt+2 set current-window-scale 2.0     # double the window size
 #d cycle deinterlace                    # toggle the deinterlacing filter
-#r add sub-pos -1                       # move subtitles up
-#R add sub-pos +1                       # move subtitles down
-#t add sub-pos +1                       # move subtitles down
+#r cycle-values video-rotate 270 180 90 0 # rotate left
+#t cycle-values video-rotate 90 180 270 0 # rotate right
```
Comment (kasper93) [inline etc/input.conf:125]:
```suggestion
#r cycle-values video-rotate 90 180 270 0 # rotate clockwise 
#R cycle-values video-rotate 270 180 90 0 # rotate counter-clockwise
```

I think rotation clockwise is more intuitive. Also no need for another key, we can use shift.

**Comment by kasper93** [comment]
This likely will be hard to convince everyone on this change.

File: etc/input.conf (hunk 1 above)
Comment (guidocella) [reply etc/input.conf:125]: So intuitive that you inverted clockwise and counter-clockwise. :P On Google search, "rotate right" image has 3x the search results of "rotate clockwise" image, but "rotate clockwise" video has more search results than "rotate right" video. Either is fine but I feel that rotate right requires less mental gymnastics to understand.

Shift+r is fine too, but t is unbound and as I wrote in the commit message I like how intuitive it is that the key to the left rotates to the left and the key to the right rotates to the right. Guess it's up to maintainers to decide which to use.

Comment (llyyr) [reply etc/input.conf:125]: r/R makes it inconsistent with the rest of the layout, even if I agree that it's better. We don't do 1/! for contrast +/-. It's 1/2...

Comment (guidocella) [reply etc/input.conf:125]: We do have it both ways with w and e/W for panscan and z and x/Z for sub-delay. We could also bind both R and t this way.

Comment (kasper93) [reply etc/input.conf:125]: That's fair point. Just for rotate r/R is common shortcut. gwenview and eog is using ctrl+r and ctrl+shift+r. irfanview is using r and shift+r.

But if we want to keep r/t it of course make sense that r rotate to the left and t to the right. And like you said it is consistent with current status, but in the same time if you want to be consistent why not bind rotate to `ctrl+left` and `ctrl+right` to not break existing r/t binding?

Anyway I always felt that default mpv key binds are using to much keys, there is not much space for custom command without modifiers.

Comment (guidocella) [reply etc/input.conf:125]: sxiv for example uses < and >, with the same logic that the left key rotates left and the right key rotates right.

ctrl+left and ctrl+right are bound to sub-seek.

I'm just gonna bind both and R t which is consistent with the sub-pos, panscan and sub-delay bindings and should please everyone. Thankfully w r and z are bound to decrements so having r decrease video-rotate/rotate left is consistent.

Comment (guidocella) [reply etc/input.conf:125]: I went with rotate clockwise/counter-clockwise which is maybe more accurate in the manual which is supposed to be more detailed, and kept the simpler and intuitive rotate right in the comment for the stats page.

**Comment by guidocella** [comment]
Alt+left and right change video-pan, so we have to choose between ctrl+alt+arrows, alt+shift+arrows and r.

**Comment by guidocella** [comment]
There would be no advantage over using r and t directly?

**Comment by guidocella** [comment]
Oh right. Well as I wrote I never liked those bindings anyway because I can't remember which moves subs up and which moves them down without trying both, and I think ctrl+up and down are much more intuitive. And it's better not to overload one key with modifier bindings which are harder to press when we can have them on separate keybindings which are still mnemonic. Alt+s and Alt+v are the only letters with Alt bindings, and it's because they have other bindings related to the same functionality on those keys, while sub-pos and rotate are completely unrelated.

**Comment by kasper93** [comment]
> I never liked those bindings anyway because I can't remember which moves subs up and which moves them down

And how would you remember what way `r` or `t` rotate? Especially that `r` is backward? Exactly the same problem. It is intuitive that `r` is a rotate and most common is clockwise. It is NOT intuitive that `t` is rotate `t is the third and fifth letter of rotate` is not something anyone would infer without memorizing the documentation, which can be already said about `sub-pos`. But after 20 years people might have already memorized `sub-pos` key binds.

> and I think ctrl+up and down are much more intuitive

Yes and the exactly same thing apply to rotate.

> we can have them on separate keybindings which are still mnemonic.

You can say that about `sxiv for example uses < and >`, but for `r` `t` it no longer is that simple to memorize.

I personal think `ctrl+r` and `ctrl+shift+r` meets all the requirement for rotate, easy to remember, established by other software, less commonly used keybind of counter-clockwise is also accessible. Not breaking old keybinds. Also should consider adding the 
```
Ctrl+Alt+r  add rotate +1
Ctrl+Alt+R add rotate -1
```
also reset keybind would be nice and this is getting crowded by keybinds.

tl;dr I don't mind new keybind whatever it is, only issue is that I don't understand why `sub-pos` has to be changed.

I know nowadays keyboards don't have numpads, but I still use it for paning video like this https://github.com/mpv-player/mpv/issues/12083#issuecomment-1666831514

**Comment by guidocella** [comment]
> And how would you remember what way `r` or `t` rotate? Especially that `r` is backward? Exactly the same problem. It is intuitive that `r` is a rotate and most common is clockwise. It is NOT intuitive that `t` is rotate `t is the third and fifth letter of rotate` is not something anyone would infer without memorizing the documentation, which can be already said about `sub-pos`. But after 20 years people might have already memorized `sub-pos` key binds.

As I wrote multiple times, by the fact that the left key rotates left and the right key rotates right, and by the fact that with all other coupled key bindings (current r and t, w and e, z and x, 1 and 2, 3 and 4, 5 and 6, 7 and 8), the left one decrements and the right one increments. t being in rotate is just a bonus secondary coincidence. But compared to these other keybindings, sub-pos requires the extra mental step that you have to figure out which between decrementing and incrementing is up and which is down, and it's faster to just ty to press both r and t IMO.

> > and I think ctrl+up and down are much more intuitive
>
> Yes and the exactly same thing apply to rotate.

Except both ctrl+left and right and alt+left and right are already bound. I don't think ctrl+alt/alt+shift bindings are easier to remember than r.

> > we can have them on separate keybindings which are still mnemonic.
>
> You can say that about `sxiv for example uses < and >`, but for `r` `t` it no longer is that simple to memorize.

I think r for rotate is easier to remember than <. Sxiv has < and > because r is bound to reloading the image.

> I personal think `ctrl+r` and `ctrl+shift+r` meets all the requirement for rotate

I guess we will need a poll if everyone prefers different bindings.

**Comment by kasper93** [comment]
> I guess we will need a poll if everyone prefers different bindings.

I'm throwing ideas, I think your approach makes sense, but the reason for already established key bind for `sub-pos` seems weak. It is not like mouse wheel seek control that was pretty commonly disliked.

> r/R makes it inconsistent with the rest of the layout

@llyyr note that there is already precedence with that style of commands. So it is not as inconsistent as you think.
```
#j cycle sub                            # switch subtitle track
#J cycle sub down                       # switch subtitle track backwards
```

**Comment by guidocella** [comment]
> the reason to replace already established key bind for `sub-pos` seems weak. It is not like mouse wheel seek control that was pretty commonly disliked.

It may just not be commonly used. I don't actually use it either (why would you need to move subs at runtime?) and I also had to change it to add `sub-margin-y` to even make it work with `--sub-align-y=top` (side note: why do `--sub-pos` docs say that `--sub-margin-y` is better, and `--sub-margin-y` docs say that you should use `--sub-pos`? lol). Another reason the r binding is bad is that letter r has nothing to do with sub-pos. I don't think we shouldn't have the best possible bindings to preserve a binding that makes no sense, is hard to use compared to up and down arrows, and may not even be used much.

> > r/R makes it inconsistent with the rest of the layout
>
> @llyyr note that there is already precedence with that style of commands. So it is not as inconsistent as you think.
>
> ```
> #j cycle sub                            # switch subtitle track
> #J cycle sub down                       # switch subtitle track backwards
> ```

Actually it wouldn't be bad to also bind k to `cycle sub down`. All letter `add` bindings have both side-by-side bindings and upper case ones (R/t, W/e, Z/x), where the left one decrements and right one increments, which I replicated. All number `add` bindings are only side-by-side keys, and even `[` `]` to `multiply speed` is a similar case.

**Comment by guidocella** [comment]
We can even bind ctrl+shift+up and down to `add secondary-sub-pos` now that it's being added. This wouldn't be possible if we cram everything on r, and it complements the ctrl and ctrl+shift left and right sub bindings well.

**Review by kasper93 — APPROVED** [review]
I agree with the change. But to merge this we need more +1 as this changes the defaults mapping. Feel free to 👍 or 👎 this comment.

**Comment by guidocella** [comment]
Opened a poll at https://github.com/mpv-player/mpv/discussions/16712 else nothing will ever get done.
